### ISSUE TITLE
"Read-Only" execution-mode

### ISSUE THREAD

**Issue by movedoa** [patrickTingen/DataDigger#9]
I have precompiled data digger, but when i start it, it still creates a few files if they are not available (user config ini, the help file, and so on)

It would be nice to have a parameter in the main config file that ensures that are no files created or written when executing datadigger.

**Comment by patrickTingen** [comment]
Hi Movedoa, sorry for the late response. I will put your request on the list for the next version, sounds like a good functionality

**Comment by patrickTingen** [comment]
What it already does now - albeit a little experimental - is something that I call "follow the ini". It means that DataDigger will use the folder that contains the file "DataDigger.ini" to store its ini files. 

So, say that you have a folder C:\progress\datadigger with all DD sources. If you place "DataDigger.ini" in c:\progress then that is the folder where additional ini files, like DataDigger-movedoa.ini will be created. 

This does not prevent DD from compiling or creating a cache and backup folder in its source folder. Could you give me a use-case, so I get some insight in how and why you want to use this?

**Comment by movedoa** [comment]
Basically i just don't want DD to write files to its source folder.
We deploy DD with our Application but there are scenarios where we only have read access in the DD directory.
We have specific folders where we have write access but these folders are very dynamic and it would be very cubersome to deploy DD to these folders.

Currently the workaround for these cases is to copy DD to the user temp directory and executing it from there but this is not very optimal.

Would it be possible to just have a setting in the ini to specify the write folder?
We deploy DD precompiled so the rcodes are no problem just all the other dynamic files.

Best case scenario for me would be if i could specify the current windows users temp folder with some additional folder, for example "%Temp%\foo\DataDigger".

With %TEMP% i would not have to change the setting in every installation because this always points to ther user temp folder which always has write access.

**Comment by patrickTingen** [comment]
The %temp% solution actually sounds very good, I will go with that. I know of at least one other user that would benefit from this. Still thinking whether it should be a setting that is available on the settings screen or just in the ini file.

**Comment by movedoa** [comment]
I mean, when you see the settings screen the files are already written so i guess it doesn't make too much sense there

**Comment by patrickTingen** [comment]
Unless you require that DD should be restarted (and possibly offer to move the settings files)

**Comment by patrickTingen** [comment]
Not sure, I do have some ideas to make it work, so don’t give up…

--
Patrick

Van: movedoa
Verzonden: woensdag 27 juni 2018 15:24
Aan: patrickTingen/DataDigger
CC: Patrick Tingen; Comment
Onderwerp: Re: [patrickTingen/DataDigger] "Read-Only" execution-mode (#9)

I don't want to bother you to much but i guess this won't make it into DD24?

**Comment by patrickTingen** [comment]
You might want to checkout the develop branche of DataDigger; I just pushed
an update that implements your request.

The idea of using env-vars was an excellent one and I adopted it, so you
can use something like %temp% in your setting. Check the DataDigger blog
<https://datadigger.wordpress.com/2018/07/13/20180713/>  for more info

You might want to test the env-var resolving. I already had some code in
place, but it gave me wrong results. I recall that I spent quite some time
developing it back then, so I wonder if my current 10-line solution is
sufficient. On my own machine it gives better results, but if you encounter
issues, I'd like to hear it.

Have a nice weekend and happy digging!

--
Patrick Tingen


Op ma 2 jul. 2018 om 08:12 schreef Patrick Tingen <patrick.tingen@gmail.com
